content-security-policy/frame-src/frame-src-same-document-meta.sub.html

```diff
@@ -1,54 +1,57 @@
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <html>
-<body></body>
-<script>
-    async_test(async test => {
+  <body></body>
+  <script>
+    async_test((test) => {
```

Comment:
So this looks mostly correct... if this test were actually an `async_test`. I'm sorry to say that this is a case where the test is actually just a `promise_test` in disguise - it's correctly creating promises and awaiting on them at every step, so I think if you revert all the changes you made (sorry!) and just replaced `async_test` with `promise_test`, it would run correctly (and it would be correct to pass an async function to it)...

Reply:
Good to know - I'm trying to remember why I didn't just go down the route of a promise_test. Can't remember if there was a problem or if I was just basing this test on the previous feedback you had given. 

In any case, I learned a lot rewriting this test so hopefully it helps me in the future :D 

Thanks for the feedback. I may not be able to work on this again for a bit, but hopefully my next one is more on-the-mark.